### PR TITLE
fixed: write all scalar anasols to VTF

### PR DESCRIPTION
**Pull request by akva2**
e.g. boussinesq was missing the temperature

### CODE REVIEW

File: src/SIM/SIMoutput.C [hunk 1]
```diff
@@ -587,10 +587,15 @@ int SIMoutput::writeGlvS1 (const Vector& psol, int iStep, int& nBlock,
           field.fillColumn(j,pSol(Vec4(*cit,time)).ptr());
         if (mySol->getScalarSol())
         {
-          cit = grid->begin_XYZ();
-          const RealFunc& sSol = *mySol->getScalarSol();
-          for (j = 1; cit != grid->end_XYZ() && haveXsol; j++, ++cit)
-            field(field.rows(),j) = sSol(Vec4(*cit,time,grid->getParam(j-1)));
+          size_t r = pSol.dim();
+          const RealFunc* psSol = mySol->getScalarSol(0);
+          for (size_t k = 0; psSol; k++) {
```
Comment (kmokstad) [inline src/SIM/SIMoutput.C]: Suggest following (cosmetic) rewrite, including the subsequent two comments:
```
  const RealFunc* psSol;
  size_t r = pSol.dim()+1;
  for (size_t k = 1; (psSol = mySol->getScalarSol(k)); k++, r++)
  {
```

File: src/SIM/SIMoutput.C [hunk 2]
```diff
@@ -587,10 +587,15 @@ int SIMoutput::writeGlvS1 (const Vector& psol, int iStep, int& nBlock,
...
+            cit = grid->begin_XYZ();
+            const RealFunc& sSol = *psSol;
+            for (j = 1; cit != grid->end_XYZ() && haveXsol; j++, ++cit)
+              field(r+k+1,j) = sSol(Vec4(*cit,time,grid->getParam(j-1)));
```
Comment (kmokstad) [inline src/SIM/SIMoutput.C]: `field(r,j) = sSol(...);`

File: src/SIM/SIMoutput.C [hunk 3]
```diff
@@ -587,10 +587,15 @@ int SIMoutput::writeGlvS1 (const Vector& psol, int iStep, int& nBlock,
...
+            psSol = mySol->getScalarSol(k+1);
```
Comment (kmokstad) [inline src/SIM/SIMoutput.C]: remove

**Comment by akva2** [comment]
jenkins build this with downstreams ifem-openfrac=19 ifem-stokes=80 ifem-navierstokes=75 please

File: src/SIM/SIMoutput.C (hunk 1 above)
Comment (akva2) [reply src/SIM/SIMoutput.C]: changed to this, with a fix (need to be k-1 in the getScalarSol call).

Comment (kmokstad) [reply src/SIM/SIMoutput.C]: Right, but then maybe better to let the loop start with k = 0 and you save one operation (since k is not used elsewhere)

**Pull request merged**